src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const separatorToString = String(separator);
```

Comment:
This part is redundant, when you add a separator to string, it will also transform to string.

Reply:
Thanks for review! I think so, but if i don't used String(separator), it not working when separator is null

Reply:
if you go from the opposite your code will be more readable